src/sql/workbench/contrib/query/browser/queryEditor.ts

```diff
@@ -95,6 +96,9 @@ export class QueryEditor extends EditorPane {
 	private _toggleSqlcmdMode: actions.ToggleSqlCmdModeAction;
 	private _exportAsNotebookAction: actions.ExportAsNotebookAction;
 
+	// taskbar separator
+	private _separator: HTMLElement;
```

Comment:
spotted a bug here, there shouldn't be a separator element private member, we need to create it every time we need one, otherwise only the last place references it will get it.
![image](https://user-images.githubusercontent.com/13777222/134419411-e19d3865-9b96-455f-bcc1-f541f6648201.png)


not introduced by you. but let's fix it.

Reply:
I'm not sure if I understand you correctly, but why do we need to create a separator every time? The task bar content is emptied, but when opening a SQL file, it should have the once created separator just like the actions right?

Reply:
it is a html element, it can only be added  to one container, from the screenshot above, you can see there is only one separator, but we are expecting a couple more as per the code.